[6.0] Test target build failure

A recent change to the Swift 6.0 toolchain has caused our tests to stop building:

```
/Users/ec2-user/jenkins/workspace/swift-testing-main-swift-6.0-macos/swift-testing/Tests/TestingTests/Test.SnapshotTests.swift:130:31: error: cannot expand accessors on variable declared with 'let'
128 | 
129 | extension Tag {
130 |   @Tag fileprivate static let testTag: Self
    |                               `- error: cannot expand accessors on variable declared with 'let'
131 | }
132 | 
```

See https://ci.swift.org/job/swift-testing-main-swift-6.0-macos/23/consoleText

--- a/Tests/TestingTests/Test.SnapshotTests.swift
+++ b/Tests/TestingTests/Test.SnapshotTests.swift
@@ -127,5 +127,5 @@ struct Test_SnapshotTests {
 }
 
 extension Tag {
-  @Tag fileprivate static let testTag: Self
+  @Tag fileprivate static var testTag: Self
 }

